core/primitives/src/crypto.rs

```diff
@@ -412,6 +431,31 @@ pub trait Pair: Sized + 'static {
 			path,
 		)
 	}
+
+	/// Return a vec filled with raw data.
+	fn to_raw_vec(&self) -> Vec<u8>;
+}
+
+/// An identifier for a type of cryptographic key.
+///
+/// 0-1024 are reserved.
+pub type KeyTypeId = u32;
+
+/// Constant key types.
+pub mod key_types {
```

Comment:
That looks pretty arbitrary. I assume it's used in the keystore, but then it should be IMO moved in the keystore crate.

Reply:
It's also going to be used in the session module, that code I copied from @rphmeier s PR